STYLE.md

```diff
@@ -0,0 +1,631 @@
+}
+```
+
+Using `%w` implies that the wrapped error may later be inspected in order to perform some specific action in response. Using `%v` implies that the wrapped error is unrecoverable. Given that the only way to extract the underlying error would be to use very fragile string matching, using `%v` clearly discourages such attempts.
+
+## Tests
+
+### Test naming
+
+Put effort into test names. Use meaningful, precise names that follow the conventions of existing code.
+
+- Follow Convention: If all tests in the same file follow the "Test(Something)(SomeFeature)" convention, for example `TestParseCommand` or `TestMergeServiceContextOverrides`, follow the same convention when adding a new test.
+- Be Precise: Are we testing parsing the layer (to check if the label is valid) or are we testing the labels themselves? If the latter, `TestParseLayer` is not as accurate as `TestLabel`.
+
+### Variable names
+
+Use `foo`, `bar`, `baz`, `qux`, `quux`, and other metasyntactic variables and placeholder names in tests.
+
+However, check existing code and be consistent: If existing tests use "alice" and "bob" or meaningful variable names like `pebble-service-name`, follow the convention. Conversely, do not use very specific names that actually mean something when it's just a generic name.
+
+### Copy-paste
+
+When adding unit tests, it's common to copy-paste an existing test which is similar and modify that. It's okay to copy-paste, but examine the naming and logic, and remove unnecessary things. Treat it as if you are writing a new test.
+
+It's also okay to use a few lines of duplicated code if it makes the test clearer, instead of creating helper functions. See [Advanced Testing with Go by Mitchell Hashimoto](https://www.youtube.com/watch?v=8hQG7QlcLBk) on this topic.
+
+### Environment variables in tests
+
+`Setenv` calls `os.Setenv(key, value)` and uses Cleanup to restore the environment variable to its original value after the test. So, instead of doing:
+
+Avoid:
+
+```go
+func TestSomething(t *testing.T) {
+    // Manually set and cleanup env var
+    originalValue := os.Getenv("FOO")
+    os.Setenv("FOO", "1")
+    defer os.Setenv("FOO", originalValue)
+
+    // Test code that uses FOO environment variable
+}
+```
+
+Prefer:
+
+```go
+func TestSomething(t *testing.T) {
+    // Let the testing package handle cleanup automatically
+    t.Setenv("FOO", "1")  // Automatically restored after test
+
+    // Test code that uses FOO environment variable
+}
+```
+
+[//]: # (### `t.Fatalf` or `t.Errorf`?)
+
+[//]: # ()
+[//]: # (If a test should not continue at a certain point, use `t.Fatalf` instead of `t.Errorf`. Do not use `t.Errorf` everywhere without thinking about it.)
+
+### Comments in tests
+
+Tests should be as self-explanatory as possible. Still, in order to quickly understand the purpose of a test, use the
+
+arange-act-assert commenting style:
+
+
+```go
+func TestCreateDirs(t *testing.T) {
+	/*
```

Comment:
Probably we should at least use line comment instead of block comment as block comment in Go is pretty rare.